app/views/course/announcements/_announcement.html.slim

```diff
@@ -0,0 +1,21 @@
+div class=("announcement #{ 'future' if announcement.valid_from > DateTime.now }")
+  h3
+    = announcement.title
+
+  - if can? :manage, Course::Announcement
+    .pull-right
+      = link_to edit_course_announcement_path(@course, announcement), class: 'btn btn-default' do
+        = fa_icon 'edit'
+
+      = link_to course_announcement_path(@course, announcement),
+                  class: 'btn btn-danger', method: :delete do
+        = fa_icon 'trash'
+
+  .timestamp
+    = announcement.valid_from.to_formatted_s(:long)
```

Comment:
no, prefer more abstractions over fewer abstractions...

if it's not a full PR, at least factor it out in this one... definitely will forget to change this next time.